lib/rdoc/yard_plugin.rb

```diff
@@ -0,0 +1,228 @@
+# Yard type parser is inspired by the following code:
+# https://github.com/lsegal/yard-types-parser/blob/master/lib/yard_types_parser.rb
+
+require_relative 'base_plugin'
+require 'strscan'
+
+module RDoc
+  class YardPlugin < BasePlugin
+    listens_to :rdoc_store_complete do |env, store|
+      store.all_classes_and_modules.each do |cm|
+        cm.each_method do |meth|
+          puts "Parsing #{meth.name}"
+          parsed_comment = Parser.new(meth.comment.text).parse
+          # meth.params = parsed_comment.param.map(&:to_s).join("\n")
+          meth.comment.text = parsed_comment.plain.join("\n")
+        end
+      end
+    end
+
+    class Parser
+      ParamData = Struct.new(:type, :name, :desc, keyword_init: true) do
+        def append_desc(line)
+          self[:desc] += line
+        end
+
+        def to_s
+          "Name: #{self[:name]}, Type: #{self[:type].map(&:to_s).join(' or ')}, Desc: #{self[:desc]}"
+        end
+      end
+      ReturnData = Struct.new(:type, :desc, keyword_init: true)
+      RaiseData = Struct.new(:type, :desc, keyword_init: true)
+      ParsedComment = Struct.new(:param, :return, :raise, :plain)
+
+      TAG_PARSING_REGEXES = {
+        param: /
+        @param\s+
+          (?:                                 # Match either of the following:
+          \[(?<type1>[^\]]+)\]\s+(?<name1>\S+)\s*(?<desc1>.*)? |  # [Type] name desc
+          (?<name2>\S+)\s+\[(?<type2>[^\]]+)\]\s*(?<desc2>.*)?   # name [Type] desc
+          )
+        /x,
+          return: /@return\s+\[(?<type>[^\]]+)\]\s*(?<desc>.*)?/,
+          raise: /@raise\s+\[(?<type>[^\]]+)\]\s*(?<desc>.*)?/
+      }
+      def initialize(comment)
+        @comment = comment
+        @parsed_comment = ParsedComment.new([], nil, [], [])
```

Comment:
It seems that we can use a local variable not an instance variable for this.

Reply:
True, thank you for pointing out.